src/astro/sql/table.py

```diff
@@ -29,8 +29,22 @@ def __init__(
     def identifier_args(self):
         return (self.schema, self.table_name) if self.schema else (self.table_name,)
 
-    def qualified_name(self):
-        return self.schema + "." + self.table_name if self.schema else self.table_name
+    def fully_qualified_name(self, conn_type, schema=None):
+        """
+        To make our SQL queries more clear in scope and to allow queries containing multiple
+        schemas or databases, we should start generating SQL that uses fully qualified names (eg.
+        schema.table in postgres and DATABASE.SCHEMA.TABLE in snowflake).
+        :param conn_type: What type of connection are we creating.
+        :param schema: an optional schema if the output_table has a schema set. Defaults to the temp schema
+        :return:
+        """
+        table_name = self.table_name
+        schema = self.schema or schema
+        if (conn_type == "postgres" or conn_type == "postgresql") and schema:
+            table_name = schema + "." + self.table_name
+        elif conn_type == "snowflake" and schema and "." not in self.table_name:
```

Comment:
Can we validate "." case when someone creates table object? and raise an exception if someone adds it? That way we can standardize this part of our code?